test/account/wallet_test.rb

```diff
@@ -68,4 +68,48 @@ def test_sign
     result = Zilliqa::Crypto::Schnorr.verify(message_hex, Zilliqa::Crypto::Signature.new(r, s), public_key)
     assert result
   end
+
+  require_relative '../contract/test_abi.rb'
+  require_relative '../contract/test_contract.rb'
+
+  def test_invalid_signature
+    private_key = Zilliqa::Crypto::KeyTool.generate_private_key
+    public_key = Zilliqa::Crypto::KeyTool.get_public_key_from_private_key(private_key)
+    address = Zilliqa::Crypto::KeyTool.get_address_from_private_key(private_key)
+    response = {
+      id: 1,
+      jsonrpc: '2.0',
+      result: {
+        balance: 888,
+        nonce: 1
+      }
+    }
+
+    provider = Minitest::Mock.new
+    wallet = Zilliqa::Account::Wallet.new(provider, {})
+    account = Zilliqa::Account::Account.new(private_key)
+    wallet.add_by_private_key(private_key)
+
+    provider.expect('testnet?', false)
+    provider.expect('GetBalance', response, [address])
+
+    100.times do |i|
+      provider.expect('testnet?', false)
+      provider.expect('GetBalance', response, [address])
+
+      tx_params = {
+        sender_pub_key: public_key,
+        amount: (10**12) * rand(1..10),
+        gas_price: rand(10..15_000),
+        gas_limit: '1',
+        to_addr: '1234567890123456789012345678901234567890',
+        code: TEST_CONTRACT.gsub('/\\', ''),
+        data: JSON.generate(ABI)
+      }
+
+      tx = Zilliqa::Account::Transaction.new(tx_params, nil)
+      sig = account.sign_transaction(tx)
+      Zilliqa::Util::Validator.signature?(sig.to_s)
```

Comment:
Is assert or something alike missed here?

Reply:
For bool result it not necessary, but i add this for read readability